### PR TITLE
Support all key algs with structured authn config

### PR DESCRIPTION
**Pull request by enj**
/kind bug
/assign aramase
/triage accepted
/priority important-soon

```release-note
Fixed an issue where a JWT authenticator configured via --authentication-config would fail to verify tokens that were not signed using RS256.
```

### CODE REVIEW

File: pkg/kubeapiserver/options/authentication.go [hunk 1]
```diff
@@ -461,6 +462,19 @@ func (o *BuiltInAuthenticationOptions) ToAuthenticationConfig() (kubeauthenticat
...
+			oidc.RS256,
+			oidc.RS384,
+			oidc.RS512,
+			oidc.ES256,
+			oidc.ES384,
+			oidc.ES512,
+			oidc.PS256,
+			oidc.PS384,
+			oidc.PS512,
+		}
```
Comment (aramase) [inline pkg/kubeapiserver/options/authentication.go:477]: Should we just have an exported function in `oidc.go` that returns keys from https://github.com/kubernetes/kubernetes/blob/7ec1a89a509906dad9fd6a4635d7bfc157b47790/staging/src/k8s.io/apiserver/plugin/pkg/authenticator/token/oidc/oidc.go#L201-L211

Then we don't have to maintain the list in 2 places.

File: pkg/kubeapiserver/options/authentication.go
Comment (aramase) [inline pkg/kubeapiserver/options/authentication.go:1]: I think we can add a test in https://github.com/aramase/kubernetes/blob/15c02f2a034d23174d7c9ef1bb61982cc36126a5/pkg/kubeapiserver/options/authentication_test.go#L448 to validate expected config with `--authentication-config` flag.

Comment (aramase) [reply pkg/kubeapiserver/options/authentication.go:1]: This is not a blocker for the PR. I can do this in a follow-up.

Comment (enj) [reply pkg/kubeapiserver/options/authentication.go:1]: Done.

File: pkg/kubeapiserver/options/authentication.go (hunk 1 above)
Comment (enj) [reply pkg/kubeapiserver/options/authentication.go]: Done.

**Review by liggitt — COMMENTED** [review]
/lgtm
/approve

**Comment by aramase** [comment]
- `pull-kubernetes-e2e-gce` failures are discussed here: https://kubernetes.slack.com/archives/C09QZ4DQB/p1707943263710419
- for `pull-kubernetes-node-e2e-containerd`, see this https://github.com/kubernetes/registry.k8s.io/issues/274

**Pull request merged**